### ISSUE TITLE
Any interest in re-homologating this with the geographic-info org version?

### ISSUE THREAD

**Issue by SteveMacenski** [ros2/unique_identifier_msgs#4]
This seems to be a ROS2 port for https://github.com/ros-geographic-info/unique_identifier

I'm guessing this was hard forked from lack of movement, now that I have write access I can rectify these. I just hate to see 3 versions of similar content 

https://github.com/ros2/unique_identifier
https://github.com/ros2/unique_identifier_msgs/
https://github.com/ros-geographic-info/unique_identifier

**Comment by jacobperron** [comment]
You can find a discussion here about the port to ROS 2: https://github.com/ros2/ros2/issues/593

IMO, it's fine for `unique_identifier_msgs` to live in the ros2 org since it is used as part of core ROS 2 packages (specifically in the implementation of Actions). Also, UUIDs are generally useful outside of the scope of geographic messages.

That being said, we've only ported the interface package (previously named `uuid_msgs`), not `unique_id`. And I think it would be fine to keep that code living in https://github.com/ros2/unique_identifier or move it to https://github.com/ros2/unique_identifier, I don't have a preference. In the latter case, I can help maintain the package. In the former case, I can delete the ros2 fork.

**Comment by robotpilot** [comment]
I also agree with @jacobperron . This message is also necessary for geographic-info, but I think it is used more widely. I think this message would be more efficient as an independent message. :)

**Comment by SteveMacenski** [comment]
I would definitely not disagree that its useful in a number of places. I don't think it really matters to me where it lives as long as we're all on the same page about it and trying to reduce redundant efforts. Having it as a ros2 repo (and making it a builtin interface) seems really reasonable to me. I just want to then have that ros2 repo port and maintain the `unique_id` headers to boot

**Comment by robotpilot** [comment]
@SteveMacenski I have been using some of the features of `ros-geographic-info` and `ROS geographic information project` since ROS 1, and now I am using some of them on ROS 2, so I would be happy to help your work. I am really grateful for your contribution. Thank you so much.

**Comment by jilaada** [comment]
Has there been any additional work on porting these packages to ROS2 Foxy since this discussion?

**Comment by jacobperron** [comment]
There hasn't been any follow-up porting the language specific packages that I'm aware of, so I went ahead and created a fork of the original repository and continued the port that @stonier started: https://github.com/ros2/unique_identifier/pull/1.

In it I've split up the c++ and python packages as per my proposal in https://github.com/ros2/ros2/issues/593#issue-374128558.

**Comment by jilaada** [comment]
@jacobperron Thanks, let me know if you need some help with the porting of the packages. In the meanwhile, do you happen to know what are people generally using to generate UUIDs in Foxy?

**Comment by jacobperron** [comment]
@jilaada I would appreciate help finishing the port. There are two outstanding tasks:

1. Fix the unit tests (failing for both C++ and Python packages)
2. Add support for time-based UUIDs (I've commented out the relevant sections of the code and left TODOs)

> do you happen to know what are people generally using to generate UUIDs in Foxy

I don't know. For ROS 2 actions, we use the `uuid` module for Python and generate random bytes in C++ (https://github.com/ros2/rclcpp/blob/31c202e325476f6e912d6d23c17d13393156dbf4/rclcpp_action/src/client.cpp#L377-L379).

**Comment by jilaada** [comment]
@jacobperron Thanks for the information, I'll take a look at the unit tests. As there are many forks and variations of the unique_identifier package, I'd like to confirm which is the current working repository and I'll begin working from there.

> generate random bytes in C++

I think I'll try this as a temp solution while this package is ported to Foxy/ROS2.
